### DISCUSSION TITLE
Loki query dashboard not validating whether a query field is empty or not

### DISCUSSION THREAD

**Discussion by SanthoshS20** [grafana/grafana#53647] [Observability]
**What happened**:
In Loki query dashboard, there is no code to validate whether a query field is empty or not. If the query field is empty, we should popup a notification to the customer to add a query or remove that query field. With empty query field, API request is processing and results in 400 Bad request. We should add validation while disabling or enabling a query field and while deleting a query field.

**What you expected to happen**:
Before performing an API request, it should validate where any one of the query field is empty. If its empty notify to the customer to delete/disable/add query. 

**How to reproduce it (as minimally and precisely as possible)**:
**Step 1:** Go the the Loki dashboard.
**Step 2:** Add a query field and add query to it.
**Step 3:** Add another query field and leave it empty.
**Step 4:** Click the run query button. 
**Step 5:** It will fetch data for the first query field and show parse error for the second query field. Look at the failed API request. 

**Anything else we need to know?**:
![image](https://user-images.githubusercontent.com/37366893/183450045-e1dfc182-712a-4610-8e1a-a48b36bb132d.png)

**Environment**:
- Grafana version: 9.0.5
- Data source type & version:
- OS Grafana is installed on:
- User OS & Browser: Windows 10 & Chrome
- Grafana plugins:
- Others:

**Comment by usmangt** [comment]
Hi @SanthoshS20 ,

Thanks for opening this case. This seems to be like a feature request (improvement) to get a notification on the UI level when the field is empty.

I will move this to our feature request section so that they can view it and get back to you with more details.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
